src/backends/ncnn/ncnnlib.cc

```diff
@@ -94,6 +93,13 @@ namespace dd
   void NCNNLib<TInputConnectorStrategy, TOutputConnectorStrategy,
                TMLModel>::init_mllib(const APIData &ad)
   {
+    bool use_fp32 = (ad.has("datatype")
+                     && ad.get("datatype").get<std::string>()
+                            == "fp32"); // default is fp16
+    _net->opt.use_fp16_packed = !use_fp32;
+    _net->opt.use_fp16_storage = !use_fp32;
```

Comment:
when loading weights from caffe, I am pretty sure they are stored as fp32, so in this case if you do not provide explicitely datatype= fp32 in API, does not everything fail ?

Reply:
ok it seems that there is an explicit conversion from fp32 to fp16 after load time and before inference if  use_weight_fp16_storage is true, as least in lstm_x86

Reply:
Thanks, this is useful. I did put/leave `fp16` as the implicit default for two reasons: when using NCNN it's usually we are running on compute-limited platform, and it seems to have been the default at least on the branch I was on, but it is the default setting in their example code snippets and benchmarks... We can revert it in case some issues do surface...